examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx

```diff
@@ -44,7 +44,17 @@ export function AppointmentActions(props: AppointmentActionsProps): JSX.Element
       await medplum.updateResource({
         ...appointment,
         status: newStatus,
+        slot: undefined,
       });
+      // If the appointment is cancelled, update the slot status to free
+      if (newStatus === 'cancelled' && appointment.slot?.[0].reference) {
+        const slotId = appointment.slot[0].reference.split('Slot/')[1];
```

Comment:
we have a helper function for this: `resolveId(reference)`

Reply:
Nice, thanks!